doc: Update ClusterMesh documentation for 1.4

tgraf opened the pull request:
Provide a getting-started guide to set up cluster mesh using SSL certificates
generated by cilium-etcd-operator. Provide examples for global services and
policy enforcement.

Provide a cross cluster policy example

Depends on https://github.com/cilium/cilium-etcd-operator/pull/35

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/6853)
<!-- Reviewable:end -->

aanm commented on Documentation/gettingstarted/clustermesh.rst:
```diff
@@ -2,115 +2,259 @@
+
+    .. code:: bash
+
+       ./generate-name-mapping.sh
```
replace these steps with:
```
kubectl patch ds -n kube-system cilium --patch "$(./generate-name-mapping.sh)"
```
where the output of `./generate-name-mapping.sh` should be:

```
spec:
  template:
    spec:
      hostAliases:
      - hostnames:
        - cluster1.mesh.cilium.io
        ip: 10.138.0.18
      - hostnames:
        - cluster2.mesh.cilium.io
        ip: 10.138.0.19
```

aanm commented on examples/kubernetes/clustermesh/global-service-example/cluster1.yaml:
```diff
@@ -0,0 +1,85 @@
+    spec:
+      containers:
+      - name: rebel-base
+        image: docker.io/nginx
```
please specify a version

tgraf replied:
Common, it's a demo pod ;-)

tgraf replied:
Changed

aanm commented on examples/kubernetes/clustermesh/global-service-example/cluster2.yaml:
```diff
@@ -0,0 +1,85 @@
+    spec:
+      containers:
+      - name: rebel-base
+        image: docker.io/nginx
```
same here

ianvernon commented on Documentation/concepts.rst:
```diff
@@ -610,6 +610,15 @@ This is typically achieved using two methods:
+Cluster Mesh
+============
+
+Cluster mesh extends the networking datapath across multiple clusters. It
```
Nit: Is the official term `ClusterMesh` or `Cluster Mesh` or `cluster mesh`? It is inconsistent throughout the document - I think `ClusterMesh` is better because it makes it sound more cilium-specific ;)

tgraf replied:
I've changed it to cluster mesh, it's what was concluded on the big servicemesh vs service mesh discussion.

ianvernon commented on Documentation/gettingstarted/clustermesh.rst:
```diff
@@ -132,3 +276,96 @@ this command inside any Cilium pod in any cluster:
+#################
+
+As addressing and network security is decoupled, network security enforcement
+automatically span across clusters. Note that Kubernetes security policies are
```
spans

ianvernon commented on examples/kubernetes/clustermesh/global-service-example/cluster1.yaml:
```diff
@@ -0,0 +1,85 @@
+    spec:
+      containers:
+      - name: x-wing-container
+        image: docker.io/cilium/json-mock
```
version here as well

joestringer commented on Documentation/gettingstarted/clustermesh.rst:
```diff
@@ -2,115 +2,259 @@
+  requirement is typically met by establishing peering or VPN tunnels between
+  the networks of the nodes of each cluster.
 
-* No encryption is performed by Cilium for the connectivity between nodes.
```
This encryption section seems to have been removed, but it's not re-added into the limitations section. In particular, are we planning to resolve encryption of traffic affected by L7 rules for v1.4?

tgraf replied:
1.4 is adding transparent encryption, the documentation is just missing at this pod. It's definitely wrong to say that Cilium is not doing any encryption at this point.

tgraf committed:
doc: Update ClusterMesh documentation for 1.4

Provide a getting-started guide to set up cluster mesh using SSL certificates
generated by cilium-etcd-operator. Provide examples for global services and
policy enforcement.

Provide a cross cluster policy example

Depends on https://github.com/cilium/cilium-etcd-operator/pull/35

Signed-off-by: Thomas Graf <thomas@cilium.io>
